app/connector/FrontendAuthConnector.scala

```diff
@@ -31,14 +35,43 @@ object GovernmentGatewayEnrolment {
   implicit val fmt = Json.format[GovernmentGatewayEnrolment]
 }
 
+case class InternalUserIdentifier(value: String) extends AnyVal
+
+object InternalUserIdentifier {
+  implicit val reads: Reads[InternalUserIdentifier] = new Reads[InternalUserIdentifier] {
+    override def reads(json: JsValue): JsResult[InternalUserIdentifier] =
+      Try((json \ "internalId").as[String])
+        .map(id => JsSuccess(InternalUserIdentifier(id)))
+        .getOrElse(JsError("Error parsing internal user identifier"))
```

Comment:
There is another way that doesn't use `Try`, it returns a JsResult that you can pattern match on
https://www.playframework.com/documentation/2.3.x/ScalaJsonCombinators